### PR TITLE
RCHAIN-2918 add rewards validate in unbonding test

### PR DESCRIPTION
**Pull request by zsluedem**
## Overview
1.  add rewards validate in unbonding test

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-2918

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: integration-tests/test/test_unbond.py [hunk 1]
```diff
@@ -81,8 +111,11 @@ def test_unbond_validator(command_line_options: CommandLineOptions, random_gener
 
         # block number 4
         # withdraw not happen yet
-        validator_1_balance = get_vault_balance(context, bootstrap_node, VALIDATOR_KEY_1.get_public_key().get_rev_address(), VALIDATOR_KEY_2, 100000, 1)
-        assert validator_1_balance < 20000000
+        validator_1_balance_before_refund = get_vault_balance(context, bootstrap_node,
+                                                              VALIDATOR_KEY_1.get_public_key().get_rev_address(),
+                                                              VALIDATOR_KEY_2, 100000, 1)
+        # the unbond process cost validator1 some phlos so the balance of validator1 is not smaller than the initial wallet amount
+        assert validator_1_balance_before_refund < validator_1_initial_wallet_amount
```
Comment (tgrospic) [inline integration-tests/test/test_unbond.py]: Did you mean _is ~~not~~ smaller than the initial wallet amount_?

**Review by tgrospic — APPROVED** [review]
LGTM, one typo in comment.

File: integration-tests/test/test_unbond.py (hunk 1 above)
Comment (zsluedem) [reply integration-tests/test/test_unbond.py]: You are right.

File: integration-tests/test/test_unbond.py [hunk 2]
```diff
@@ -81,8 +111,11 @@ def test_unbond_validator(command_line_options: CommandLineOptions, random_gener
 
         # block number 4
         # withdraw not happen yet
-        validator_1_balance = get_vault_balance(context, bootstrap_node, VALIDATOR_KEY_1.get_public_key().get_rev_address(), VALIDATOR_KEY_2, 100000, 1)
-        assert validator_1_balance < 20000000
+        validator_1_balance_before_refund = get_vault_balance(context, bootstrap_node,
+                                                              VALIDATOR_KEY_1.get_public_key().get_rev_address(),
+                                                              VALIDATOR_KEY_2, 100000, 1)
+        # the unbond process cost validator1 some phlos so the balance of validator1 is smaller than the initial wallet amount
+        assert validator_1_balance_before_refund < validator_1_initial_wallet_amount
```
Comment (Isaac-DeFrain) [inline integration-tests/test/test_unbond.py]: Is there no way to get the precharge amount (say `validator_1_initial_precharge_amount`) and assert the exact difference, i.e.
`assert validator_1_balance_before_refund == validator_1_initial_wallet_amount - validator_1_initial_precharge_amount`?

Comment (Isaac-DeFrain) [inline integration-tests/test/test_unbond.py]: Shouldn't we also check validator_1's balance after refund?

File: integration-tests/test/test_unbond.py [hunk 3]
```diff
@@ -105,10 +142,15 @@ def test_unbond_validator(command_line_options: CommandLineOptions, random_gener
         wait_for_node_sees_block(context, bootstrap_node, b8)
 
         # block number 9
-        validator_1_balance = get_vault_balance(context, bootstrap_node, VALIDATOR_KEY_1.get_public_key().get_rev_address(), VALIDATOR_KEY_2, 100000, 1)
-        assert validator_1_balance < 20000000
+        validator_1_balance_before_refund = get_vault_balance(context, bootstrap_node,
+                                                              VALIDATOR_KEY_1.get_public_key().get_rev_address(),
+                                                              VALIDATOR_KEY_2, 100000, 1)
+        assert validator_1_balance_before_refund < validator_1_initial_wallet_amount
```
Comment (Isaac-DeFrain) [inline integration-tests/test/test_unbond.py]: Same as above

File: integration-tests/test/test_unbond.py [hunk 4]
```diff
@@ -105,10 +142,15 @@ def test_unbond_validator(command_line_options: CommandLineOptions, random_gener
...
         # block number 10
         # withdraw happen in block number 9, result get in block 10
-        validator_1_balance = get_vault_balance(context, bootstrap_node, VALIDATOR_KEY_1.get_public_key().get_rev_address(), VALIDATOR_KEY_2, 100000, 1)
-        assert validator_1_balance > 50000000
\ No newline at end of file
+        # get the bonding amount and the reward amount
+        validator_1_balance = get_vault_balance(context, bootstrap_node,
+                                                VALIDATOR_KEY_1.get_public_key().get_rev_address(), VALIDATOR_KEY_2,
+                                                100000, 1)
+        assert validator_1_balance == validator_1_balance_before_refund + validator_1_initial_bonding_amount + rewards_of_v1
```
Comment (Isaac-DeFrain) [inline integration-tests/test/test_unbond.py]: Why should
`validator_1_balance == validator_1_balance_before_refund + validator_1_initial_bonding_amount + rewards_of_v1` 
be true as opposed to
`validator_1_balance == validator_1_balance_after_refund + validator_1_initial_bonding_amount + rewards_of_v1`?

**Review by Isaac-DeFrain — COMMENTED** [review]
I left a few questions.

File: integration-tests/test/test_unbond.py (hunk 2 above)
Comment (zsluedem) [reply integration-tests/test/test_unbond.py]: I think I might name a bad name for the variable `validator_1_balance_before_refund `. The `refund` here means bonding refund not deploy refund

**Review by Isaac-DeFrain — APPROVED** [review]

**Pull request merged**